src/wallet/external_signer_scriptpubkeyman.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparams.h>
+#include <wallet/external_signer.h>
+#include <wallet/external_signer_scriptpubkeyman.h>
+
+
+bool ExternalSignerScriptPubKeyMan::SetupDescriptor(std::unique_ptr<Descriptor> desc)
+{
+    LOCK(cs_desc_man);
+    assert(m_storage.IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS));
+    assert(m_storage.IsWalletFlagSet(WALLET_FLAG_EXTERNAL_SIGNER));
+
+    int64_t creation_time = GetTime();
+
+    // Make the descriptor
+    WalletDescriptor w_desc(std::move(desc), creation_time, 0, 0, 0);
+    m_wallet_descriptor = w_desc;
+
+    // Store the descriptor
+    WalletBatch batch(m_storage.GetDatabase());
+    if (!batch.WriteDescriptor(GetID(), m_wallet_descriptor)) {
+        throw std::runtime_error(std::string(__func__) + ": writing descriptor failed");
+    }
+
+    // TopUp
+    TopUp();
+
+    m_storage.UnsetBlankWalletFlag(batch);
+    return true;
+}
+
+ExternalSigner ExternalSignerScriptPubKeyMan::GetExternalSigner() {
+#ifdef ENABLE_EXTERNAL_SIGNER
```

Comment:
From what I can see, `GetExternalSigner()` is only called from code that is inside `ENABLE_EXTERNAL_SIGNER` #idfefs. So why can't we wrap this whole function in an #ifdef, and drop the need for the `throw()`? I think the same goes for most of this file, and some other functions in this PR. It seems weird that when external signing is not enabled, we'd still be compiling functions that aren't called from anywhere, and who's purpose is to throw a runtime error to tell you about external signing and it's need of Boost Process.

Reply:
If turns out I can put the entire `ExternalSignerScriptPubKeyMan` inside `#ifdef`. The most important `throw()`s are already in `wallet.cpp`.